bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -195,7 +195,26 @@ private String getJsContent() {
                 headLibResource = Utils.resolveResource(resolver, headLibPath);
                 if (headLibResource == null) {
                     LOG.trace("No custom headlib for resource type {}.", resourceType);
-                    continue;
+
+                    // Get resource from the resource type.
+                    Resource coreResource = slingRequest.getResourceResolver().getResource(resourceType);
```

Comment:
this won't work for anonymous users, they won't have access to the /apps/ tree. you'll need to use the Utils method used above

Reply:
You will also need make sure the service user you're using has access to the actual component node as well, so the rep:policy will need to be updated at /apps/_rep_policy.xml

Reply:
I changed it to use _Utils.resolveResource_, unfortunately I'm constantly receiving _null_ with that, and because of what you commented I think it is a permission issue. I created a user like this
 
![image](https://user-images.githubusercontent.com/57115670/68446249-d7329400-01a1-11ea-93d3-c8bf2a0c295b.png)

But it is still not working, I would appreciate if you can point me the right direction to solve it, am I missing something?

Also I would like to know if you have a dev environement to match my local configurations, thanks!

Reply:
As mentioned, you will need to update the permissions of the service user. look at the node in the repo at `/config/src/content/jcr_root/apps/_rep_policy.xml`. You'll see the current added permissions. We will need to come up with the proper policy to allow this service user to read the actual component nodes.

Reply:
@id-keenan done! Please let me know what you think or if there may be something else, thanks.